### ISSUE TITLE
As a user, I can hide the palette and the minimap

### ISSUE THREAD

**Issue by andreasgeier** [camunda/camunda-modeler#2903]
__Is your feature request related to a problem? Please describe.__

UX feedback: As a power-user, I rarely need the palette and would like to be able to hide it, e.g. with a shortcut key similar to the properties panel, in order to have maximum screen real estate on low screen resolutions as well as better focus on the model and nothing that blocks the view to the start of my process, e.g. when opening it or after using Ctrl+1 to 'Zoom to Fit Diagram'. Alternatively, one could also put the palette into a toolbar  outside the canvas or floating at the bottom left. Since it is used so rarely, it feels rather like a UX annoyance obstructing the view than a tool so critical that it needs floating arround all the time.
Also, they want better focus on the diagram, for instance, in an implementation phase where the palette is not needed.

__Describe the solution you'd like__

For an uninterrupted view of the BPMN and DMN model, we should allow the user to hide tools that overlay the canvas, like the palette and mini-map. This could be done via keyboard shortcut and App menu > Window > Toggle Palette / Toggle Mini-Map

__Alternative__

Enable docking of the palette (would need further investigation).

__Additional context__

Based on UX feedback from Luminess, reported by @falko via [Slack](https://camunda.slack.com/archives/CKGH9LR40/p1649352353773439)

**Comment by nikku** [comment]
Even more so as the basics will be fully keyboard accessible (`n` / `a`) to create/append anything in the future. 

I think as this is an expert feature, hiding it behind a keyboard shortcut would be appropriate, too. I'd rather wait for data before we decide to hide it entirely; it is just too common of a feature ("escape latch") for people building things.

**Comment by andreasgeier** [comment]
> I'd rather wait for data before we decide to hide it entirely

We should not remove it entirely. Also, I would not hide it by default. This request is about toggling which would add value without risk.

**Comment by christian-konrad** [comment]
Keep this in backlog until append/create anything is evaluated. We need to see data on usage of the palette vs. create anything menu from the palette vs. context pad append anything menu. Then may reconsider this

**Comment by andreasgeier** [comment]
> We need to see data on usage of the palette vs. create anything menu from the palette vs. context pad

I don't see a strict correlation between toggling palette and context pad/append anything. For me, this is just an easy way to give the user more control over the Modeler appearance. Side panel and bottom panel can be toggled, so why not palette and mini-map, too?

Also, let's not forget there are more usage scenarios besides diagram design, for instance just _reading_ it. And, there is a reason why the _token simulation_ hides the palette when it's active.

**Comment by nikku** [comment]
Agreed with @andreasgeier. We can implement this easily if we find an elegant pattern that fits into the existing usage behavior. It is a classic "configurable power tool" feature.
